RELEASE.md

```diff
@@ -1,31 +1,68 @@
 # How to release Kuadrant Operator
 
-## Process
+## Kuadrant Operator Release
 
 To release a version _“v0.W.Z”_ of Kuadrant Operator in GitHub and Quay.io, follow these steps:
 
-1. Kuadrant dependencies need to be released first:
+### Release file format.
```

Comment:
So we are abandoning  the "release button" in favour of creating branch + release file? I reckoned it was very convenient to have a workflow such as https://github.com/Kuadrant/limitador-operator/blob/main/.github/workflows/release.yaml#L4

Reply:
I'd say is good to keep both, but not relying on the file to be a requisite for the release... those are 3 extra steps for releasing instead of 1.

Reply:
And nothing prevent us for making mistakes in the file neither xD

Reply:
I don't think this is an abandonment of the "release button", but more stepping back to reevaluate. Some of the manual local steps can, and possible should be automated again in the future. The main focus is how do we reduce the chance of errors during the release, but also give us an escape route for when the error happen. That escape route would be the PR review.

Reply:
in that case, this PR shouldn't remove the automated version, but mention both alternatives